### PR TITLE
Water and lava flow only for regions

### PR DESCRIPTION
**Pull request by arturekdev**
https://github.com/FunnyGuilds/FunnyGuilds/issues/1199

### CODE REVIEW

File: src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java [hunk 1]
```diff
@@ -17,28 +17,9 @@
...
-import net.dzikoysk.funnyguilds.listener.region.BucketAction;
-import net.dzikoysk.funnyguilds.listener.region.EntityExplode;
-import net.dzikoysk.funnyguilds.listener.region.EntityPlace;
-import net.dzikoysk.funnyguilds.listener.region.GuildHeartProtectionHandler;
-import net.dzikoysk.funnyguilds.listener.region.HangingBreak;
-import net.dzikoysk.funnyguilds.listener.region.HangingPlace;
-import net.dzikoysk.funnyguilds.listener.region.PlayerCommand;
-import net.dzikoysk.funnyguilds.listener.region.PlayerInteract;
-import net.dzikoysk.funnyguilds.listener.region.PlayerMove;
-import net.dzikoysk.funnyguilds.listener.region.PlayerRespawn;
```
Comment (CDFN) [inline src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java:41]: Set up your IDE so it doesn't delete imports like these, too big diff imo

File: src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java [hunk 2]
```diff
@@ -58,31 +39,32 @@
...
     private String mainVersion;
-
-    private final File pluginConfigurationFile  = new File(this.getDataFolder(), "config.yml");
-    private final File messageConfigurationFile = new File(this.getDataFolder(), "messages.yml");
-    private final File pluginDataFolderFile     = new File(this.getDataFolder(), "data");
-
-    private FunnyGuildsLogger      logger;
-    private PluginConfiguration    pluginConfiguration;
-    private MessageConfiguration   messageConfiguration;
-    private ConcurrencyManager     concurrencyManager;
```
Comment (CDFN) [inline src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java:72]: Don't change codestyle

File: src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java [hunk 3]
```diff
@@ -58,31 +39,32 @@
 public class FunnyGuilds extends JavaPlugin {
 
     private static FunnyGuilds funnyguilds;
-
```
Comment (CDFN) [inline src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java:61]: Don't change codestyle

File: src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java [hunk 4]
```diff
@@ -58,31 +39,32 @@
...
+    private MessageConfiguration messageConfiguration;
+    private ConcurrencyManager concurrencyManager;
     private DynamicListenerManager dynamicListenerManager;
 
     private volatile BukkitTask guildValidationTask;
     private volatile BukkitTask tablistBroadcastTask;
     private volatile BukkitTask rankRecalculationTask;
 
-    private DataModel                    dataModel;
-    private DataPersistenceHandler       dataPersistenceHandler;
```
Comment (CDFN) [inline src/main/java/net/dzikoysk/funnyguilds/FunnyGuilds.java:80]: Don't change codestyle

**Review by CDFN — COMMENTED** [review]
Don't change codestyle that drastically.

File: src/main/java/net/dzikoysk/funnyguilds/listener/BlockFlow.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockFromToEvent;
+
+public class BlockFlow implements Listener {
+
+    @EventHandler
+    public void blockFlow(BlockFromToEvent event) {
+
+        Block block = event.getBlock();
+        if (block.getType() == Material.LAVA || block.getType() == Material.WATER) {
```
Comment (CDFN) [inline src/main/java/net/dzikoysk/funnyguilds/listener/BlockFlow.java:17]: Maybe use https://papermc.io/javadocs/paper/1.16/org/bukkit/block/Block.html#isLiquid-- instead? I'm not entirely sure if this method is supported in legacy versions, e.g. 1.8.8

Comment (arturekdev) [reply src/main/java/net/dzikoysk/funnyguilds/listener/BlockFlow.java:17]: Good idea

**Review by dzikoysk — APPROVED** [review]

**Pull request merged**